sdk/core/Azure.Core/src/Pipeline/HttpPipeline.cs

```diff
@@ -109,12 +109,14 @@ public HttpMessage CreateMessage(RequestContext? context)
         /// <returns>The message.</returns>
         public HttpMessage CreateMessage(RequestContext? context, ResponseClassifier? classifier = default)
         {
-            HttpMessage message = CreateMessage();
-            if (classifier != null)
+            Request request = CreateRequest();
+            HttpMessage message = new(request, classifier ?? ResponseClassifier);
+
+            if (context != null)
             {
-                message.ResponseClassifier = classifier;
+                message.ApplyRequestContext(context, classifier);
```

Comment:
why did you have to change this line of code? Is the property setter not backward compatible?

Reply:
No, it's because I made context non-nullable in this method signature in HttpMessage because I thought it was cleaner: https://github.com/Azure/azure-sdk-for-net/pull/42262/files#diff-4328bf1786a3d4c38103fc67fe893d7d9f52fa959817b636c19d59dd87531179R97

I can put it back if it's a concern.

I'm also looking at the Apply change we discussed offline - I may update this as part of that if we decide to do that.

Reply:
Updated per offline discussion.